segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DataFrameReader.java

```diff
@@ -110,7 +112,7 @@ public void close() {
 
                     this.lastReadSequenceNumber = seqNo;
                     return new ReadResult<>(logItem, segments);
-                } catch (SerializationException ex) {
+                } catch (IOException ex) {
```

Comment:
Yes. Previously I was wrapping some `IOException`s into `SerializationException` and this would catch them all. In order to preserve correctness, I need to catch `IOException` here, which includes `EOFException` and other stuff that is still `DataCorruptionException`. Note that network exceptions will be thrown by BookKeeper and those are not `IOException`s, so we won't be throwing DCE for such an issue.